engine/src/main/java/io/zeebe/engine/processor/workflow/variable/UpdateVariableDocumentProcessor.java

```diff
@@ -37,7 +37,7 @@ public void onCommand(
     final VariableDocumentRecord record = command.getValue();
 
     final ElementInstance scope = elementInstanceState.getInstance(record.getScopeKey());
-    if (scope == null) {
+    if (scope == null || scope.isTerminating() || scope.isInFinalState()) {
```

Comment:
Oh yeah, that makes sense, my bad :see_no_evil:

Reply:
Found by accident :sweat_smile: